src/lib/containers/HasAccessV2.tsx

```diff
@@ -0,0 +1,321 @@
+        } else {
+          setAccessRequirements(requirements)
+          setDisplayAccessRequirement(true)
+        }
+      },
+    )
+  }
+
+  // Show Access Requirements
+  const AccessRequirements = () => {
+    const { entityId } = props
+    if (!restrictionInformation) {
+      // loading
+      return <></>
+    }
+    const hasUnmetAccessRequirement =
+      restrictionInformation?.hasUnmetAccessRequirement
+    const restrictionLevel = restrictionInformation?.restrictionLevel
+    let linkText = ''
+
+    if (hasUnmetAccessRequirement) {
+      linkText = 'Request Access'
+    } else if (RestrictionLevel.OPEN === restrictionLevel) {
+      // they need to sign in
+      return <></>
+    } else {
+      linkText = 'View Terms'
+    }
+    return (
+      <>
+        <a
+          style={{
+            fontSize: '14px',
+            cursor: 'pointer',
+            marginLeft: '5px',
+            verticalAlign: 'middle',
+          }}
+          className={props.className}
+          onClick={handleGetAccess}
+        >
+          {linkText}
+        </a>
+        {displayAccessRequirement && (
+          <AccessRequirementList
+            entityId={entityId}
+            accessRequirementFromProps={accessRequirements}
+            renderAsModal={true}
+            onHide={() => {
+              setDisplayAccessRequirement(false)
+            }}
+          />
+        )}
+      </>
+    )
+  }
+
+  if (fileHandleDownloadType === undefined) {
+    // note, this can't be "if (!downloadType)" since DownloadTypeEnum has a 0 value (which is falsy)
+    // loading
+    return <></>
+  }
+  const iconContainer =
+    fileHandleDownloadType ===
+    FileHandleDownloadTypeEnum.AccessBlockedToAnonymous ? (
+      <button
+        type="button"
+        className={SRC_SIGN_IN_CLASS}
+        onClick={ev => {
+          if (ev.isTrusted) {
+            /*
+                There is a tricky problem - 
+                The portals listens to click events for buttons with the class SRC_SIGN_IN_CLASS set, it listens to this event
+                so that it can display the login modal.
+
+                This button has an svg inside of it which is problematic because more often than not clicking this button will 
+                instead click that svg. The event listener in the portals will break as a result.
+
+                Though the svg may get the actual click event, because of event bubbling this button will get its onClick called.
+                Once onClick is called we can manually dispatch an event off of this button. This does pose a problem, we end up in a 
+                infinite loop because this button keeps disptaching click events, so we can use the isTrusted to recognize if onClick was
+                triggered programmatically or by user click. Lastly, using { bubbles: true } ensures the event bubbles up to the document level.
+
+              */
+            const clickEvent = new MouseEvent('click', { bubbles: true })
```

Comment:
Interesting, nice workaround here.  Is there a better way to trigger the sign-in modal on the portals side (instead of using SRC_SIGN_IN_CLASS)?

Reply:
Probably--this code and comment exists in HasAccess v1, so I didn't touch it.

I think it would make more sense to pass a callback called `onShowLogin` via `SynapseContext`. Each app using SRC could trigger showing the login screen however is most appropriate